From d9af7ea3f4397889a3d18d27523259a252ca3911 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Wed, 18 Jan 2023 10:28:48 +0100
Subject: [PATCH 018/120] hwspinlock: support hwlock-cells = 2 in default
 binding

Since Linux commits "hwspinlock: allow sharing of hwspinlocks",
the simple translation function change to support 2 parameters.

If a device registers to the framework with #hwlock-cells = 2, then
the second parameter of the 'hwlocks' DeviceTree property defines
whether an hwlock is requested for an exclusive or a shared usage.

Even if this parameter is not supported in U-Boot hwspinlock uclass,
this patch adds support of device tree with this new parameter.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git b43d01f23aac4760aa93978a6bb940343bb38995 hwspinlock: support hwlock-cells = 2 in default binding]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I99ade8215b846d9a67085f988800071754f6cd8f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/285578
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 drivers/hwspinlock/hwspinlock-uclass.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hwspinlock/hwspinlock-uclass.c b/drivers/hwspinlock/hwspinlock-uclass.c
index ea93efc97df..91ca95c2bc2 100644
--- a/drivers/hwspinlock/hwspinlock-uclass.c
+++ b/drivers/hwspinlock/hwspinlock-uclass.c
@@ -23,7 +23,7 @@ hwspinlock_dev_ops(struct udevice *dev)
 static int hwspinlock_of_xlate_default(struct hwspinlock *hws,
 				       struct ofnode_phandle_args *args)
 {
-	if (args->args_count > 1) {
+	if (args->args_count > 2) {
 		debug("Invalid args_count: %d\n", args->args_count);
 		return -EINVAL;
 	}
-- 
2.47.2

